app/javascript/bundles/Dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -3,25 +3,34 @@ import PropTypes from "prop-types";
 import _ from "lodash";
 import { WidthProvider, Responsive } from "react-grid-layout";
 
-const ResponsiveReactGridLayout = WidthProvider(Responsive);
+import * as api from '../../Api/layout_api';
+import * as config from '../config.js';
 
 import "./dashboard.scss";
 import "react-grid-layout/css/styles.css";
 import "react-resizable/css/styles.css";
 
+const ResponsiveReactGridLayout = WidthProvider(Responsive);
+
 class Dashboard extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      layout: []
+      layout: [],
+      response: ""
     }
 
     this.onLayoutChange = this.onLayoutChange.bind(this);
   }
 
+  componentDidMount() {
+    api.load_layout(this);
+  }
+
   onLayoutChange(layout) {
-    this.setState({ layout });
+    this.setState({ layout: layout });
+    api.save_layout(this.state.layout);
```

Comment:
setState is async. so can't be sure that in next line its already updated state.

https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value